# paths.py

# Genome and annotations
REFERENCE_FASTA = "/costellolab/data3/sermare/arriba_hg38/hg38.fa"
CYTOBAND_FILE = "~/cytoBand.txt"

# Purity and metadata
PURITY_SCORES = "~/jupyter-notebooks/14112024_purity_scores.csv"
GTEX_FUSIONS_OLD = "/c4/home/sermare/jupyter-notebooks/old_notebooks/GTEX_fusions.csv"
GTEX_FUSIONS_NEW = "/c4/home/sermare/random_tsv_files/GTEX_TCGA_fusion_inspector.csv"
COSMIC_TABLE = '/c4/home/sermare/fusioncharacterization_gliomas/fcg/tables/Cosmic_Fusion_v101_GRCh38.tsv'

# Gene annotations
CENSUS_ONCOGENES = "~/random_tsv_files/Census_allWed Mar 15 21_17_28 2023.tsv"
TF_LIST = "~/random_tsv_files/TF_names_v_1.01.txt"
KINASE_MAP = "~/random_tsv_files/Map/globalKinaseMap.csv"

# Fusion Data Files (input)
SRA_FUSION_DIRECTORY = "/costellolab/data3/sermare/datasets/SRA/"
SRA_FUSION_DIRECTORY_2 = "/costellolab/data3/sermare/datasetsSRA/"

# Outputs / processed files
AGGREGATED_FUSIONS = "/c4/home/sermare/updated_tables_fusions/congregated_table.csv"
PATIENT_ANNOTATIONS = "/c4/home/sermare/updated_tables_fusions/patients_without_annotations.csv"
EXPORTED_SEQUENCES = "/c4/home/sermare/jupyter-notebooks/sequences_costello_subset.csv"
