{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3e724898",
   "metadata": {},
   "source": [
    "### Loading dataframes and packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fab54b4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys, os\n",
    "os.path.dirname(sys.executable)\n",
    "sys.path.append('/c4/home/sermare/fusioncharacterization_gliomas/')\n",
    "\n",
    "from fcg.common_imports import *\n",
    "import importlib\n",
    "from fcg import data_utils\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "682c3929",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = pd.read_csv('/c4/home/sermare/tables_fusion_characterization/60_patients_metadata.csv')\n",
    "gary_table = pd.read_csv('/c4/home/sermare/jupyter-notebooks/Gary_patient_samples.csv', header = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2e5d8760",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total reads processed across all files: 100409070950\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/sermare/ipykernel_2215568/1399904705.py:66: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df_jocostello['file_name']  = df_jocostello['file'].str.replace('/trimmed_fastqs', '').str.replace('_trimming_report.txt', '')\n",
      "/scratch/sermare/ipykernel_2215568/1399904705.py:67: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df_jocostello['full_path']  = df_jocostello['file'].str.replace('/trimmed_fastqs', '').str.replace('_trimming_report.txt', '')\n"
     ]
    }
   ],
   "source": [
    "samples_ran_arriba = glob.glob('/costellolab/data1/sermare/fusion_hg38/230502024/*/*trimming_report.txt')\n",
    "\n",
    "def extract_file_path_version(fp):\n",
    "    \"\"\"\n",
    "    Looks for patterns like 'V3', 'V9-2', etc. in file_path.\n",
    "    We'll capture them as 'v3', 'v9-2', etc. to match the style used in SF#unique.\n",
    "    \"\"\"\n",
    "    # Basic approach: look for 'V' + digits optionally followed by '-digits' etc.\n",
    "    # e.g. \"P533SF12827-V9_S290_\" => we want \"v9\"\n",
    "    # If you have multiple dashes like \"V9-2\", you'd handle that as well.\n",
    "    # Adjust this regex if you have more complicated patterns:\n",
    "    m = re.search(r'V(\\d+(-\\d+)?)', fp, flags=re.IGNORECASE)\n",
    "    if m:\n",
    "        # group(1) might be e.g. \"9\", or \"9-2\", etc.\n",
    "        return m.group(1)  # e.g. \"v9\" or \"v9-2\"\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "# Store total reads processed per file\n",
    "total_reads = {}\n",
    "\n",
    "for file in samples_ran_arriba:\n",
    "    with open(file, 'r') as f:\n",
    "        for line in f:\n",
    "            if \"Total reads processed:\" in line:\n",
    "                # Extract the number using regex\n",
    "                match = re.search(r\"Total reads processed:\\s+([\\d,]+)\", line)\n",
    "                if match:\n",
    "                    reads = int(match.group(1).replace(\",\", \"\"))  # Remove commas for conversion\n",
    "                    total_reads[file] = reads\n",
    "\n",
    "# Compute the sum of all reads processed\n",
    "total_reads_sum = sum(total_reads.values())\n",
    "\n",
    "# Print results\n",
    "print(f\"Total reads processed across all files: {total_reads_sum}\")\n",
    "\n",
    "df = pd.DataFrame(list(total_reads.items()), columns=[\"file\", \"processed_Arriba\"])\n",
    "df['file']  = df['file'].str.upper()\n",
    "df['file'] = [x.split('/')[-1].replace('.fastq.gz_trimming_report.txt','').split('_L')[0] for x in df['file']]\n",
    "\n",
    "samples_ran_arriba = glob.glob('/costellolab/data2/jocostello/rna_all/*/trimmed_fastqs/*trimming_report.txt')\n",
    "\n",
    "# Store total reads processed per file\n",
    "total_reads_jocostello = {}\n",
    "\n",
    "for file in samples_ran_arriba:\n",
    "    with open(file, 'r') as f:\n",
    "        for line in f:\n",
    "            if \"Total reads processed:\" in line:\n",
    "                # Extract the number using regex\n",
    "                match = re.search(r\"Total reads processed:\\s+([\\d,]+)\", line)\n",
    "                if match:\n",
    "                    reads = int(match.group(1).replace(\",\", \"\"))  # Remove commas for conversion\n",
    "                    total_reads_jocostello[file] = reads    \n",
    "            elif \"Processed reads:\" in line:\n",
    "                # Extract the number using regex\n",
    "                match = re.search(r\"Processed reads:\\s+([\\d,]+)\", line)\n",
    "                if match:\n",
    "                    reads = int(match.group(1).replace(\",\", \"\"))  # Remove commas for conversion\n",
    "                    total_reads_jocostello[file] = reads\n",
    "\n",
    "df_jocostello = pd.DataFrame(list(total_reads_jocostello.items()), columns=[\"file\", \"processed_reads\"])\n",
    "df_jocostello['Parent_Folder'] = [x.split('/')[-3] for x in df_jocostello['file']]\n",
    "\n",
    "df_jocostello['file_name']  = df_jocostello['file'].str.replace('/trimmed_fastqs', '').str.replace('_trimming_report.txt', '')\n",
    "df_jocostello['full_path']  = df_jocostello['file'].str.replace('/trimmed_fastqs', '').str.replace('_trimming_report.txt', '')\n",
    "df_jocostello['file_name']  = [x.split('/')[-1] for x in df_jocostello['file_name']]\n",
    "\n",
    "df_jocostello['file']  = df_jocostello['file'].str.upper()\n",
    "df_jocostello['file'] = [x.split('/')[-1].replace('.fastq.gz_trimming_report.txt','').split('_L')[0].split('_R')[0] for x in df_jocostello['file']]\n",
    "\n",
    "df_jocostello['Patient'] = df_jocostello['file'].apply(data_utils.extract_p_number)\n",
    "df_jocostello['SF#'] = df_jocostello['file'].apply(data_utils.extract_sf_number)\n",
    "df_jocostello = df_jocostello[df_jocostello['Patient'].notna()]\n",
    "\n",
    "df_jocostello['V'] = df_jocostello['file_name'].apply(extract_file_path_version)\n",
    "df_jocostello = df_jocostello[df_jocostello['Patient'].notna()]\n",
    "df_jocostello = df_jocostello.sort_values(by=['Parent_Folder','Patient','V'])\n",
    "\n",
    "df_jocostello = pd.merge(left = df_jocostello, right = gary_table[['SF','Histology','Tumor']], left_on = 'SF#', right_on ='SF', how = 'left')[['Patient','Tumor','Histology','SF#','Parent_Folder','file','file_name','processed_reads','full_path']]\n",
    "\n",
    "merged_dataframe = pd.merge(left=df_jocostello, right=df, left_on='processed_reads', right_on='processed_Arriba', how='outer')\n",
    "merged_dataframe['identical_processed'] = merged_dataframe['processed_Arriba'] == merged_dataframe['processed_reads']\n",
    "merged_dataframe = merged_dataframe.drop_duplicates()\n",
    "\n",
    "# df_jocostello.to_csv('COSTELLO_RNA_Metadata.csv', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8eca94c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "multisampled_rna = df_jocostello.groupby(['Patient','SF#'])['file'].size().reset_index().sort_values(by ='file').tail(76)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5ff68d76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Histology</th>\n",
       "      <th>SF#</th>\n",
       "      <th>Parent_Folder</th>\n",
       "      <th>file</th>\n",
       "      <th>file_name</th>\n",
       "      <th>processed_reads</th>\n",
       "      <th>full_path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>P303</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Astro</td>\n",
       "      <td>SF10809</td>\n",
       "      <td>CH04</td>\n",
       "      <td>P303SF10809V1_S87</td>\n",
       "      <td>p303SF10809v1_S87_R1_001.fastq.gz</td>\n",
       "      <td>26641455</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CH04/p30...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>P375</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Astro</td>\n",
       "      <td>SF11136</td>\n",
       "      <td>CH04</td>\n",
       "      <td>P375SF11136V1_S95</td>\n",
       "      <td>P375SF11136v1_S95_R2_001.fastq.gz</td>\n",
       "      <td>30283276</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CH04/P37...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>P302</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF10788</td>\n",
       "      <td>CHJC05</td>\n",
       "      <td>P302SF10788V1_S1</td>\n",
       "      <td>p302SF10788v1_S1_R1_001.fastq.gz</td>\n",
       "      <td>28842166</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC05/p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>P327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF10932</td>\n",
       "      <td>CHJC05</td>\n",
       "      <td>P327SF10932V1_S22</td>\n",
       "      <td>p327SF10932v1_S22_R1_001.fastq.gz</td>\n",
       "      <td>30786841</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC05/p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>P340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF11055</td>\n",
       "      <td>CHJC05</td>\n",
       "      <td>P340SF11055V1_S32</td>\n",
       "      <td>p340SF11055v1_S32_R1_001.fastq.gz</td>\n",
       "      <td>34255351</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC05/p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>P413</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF11173</td>\n",
       "      <td>CHJC05</td>\n",
       "      <td>P413SF11173V1_S11</td>\n",
       "      <td>p413SF11173v1_S11_R1_001.fastq.gz</td>\n",
       "      <td>45605532</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC05/p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>P452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF11644</td>\n",
       "      <td>CHJC06</td>\n",
       "      <td>P452SF11644V1_S27</td>\n",
       "      <td>p452SF11644v1_S27_R1_001.fastq.gz</td>\n",
       "      <td>57408319</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC06/p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>P532</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SF13060</td>\n",
       "      <td>CHJC17</td>\n",
       "      <td>P532SF13060-V1_S29</td>\n",
       "      <td>P532SF13060-v1_S29_R1_001.fastq.gz</td>\n",
       "      <td>44475151</td>\n",
       "      <td>/costellolab/data2/jocostello/rna_all/CHJC17/P...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Patient    Tumor Histology      SF# Parent_Folder                file  \\\n",
       "23     P303  Primary     Astro  SF10809          CH04   P303SF10809V1_S87   \n",
       "39     P375  Primary     Astro  SF11136          CH04   P375SF11136V1_S95   \n",
       "103    P302      NaN       NaN  SF10788        CHJC05    P302SF10788V1_S1   \n",
       "123    P327      NaN       NaN  SF10932        CHJC05   P327SF10932V1_S22   \n",
       "143    P340      NaN       NaN  SF11055        CHJC05   P340SF11055V1_S32   \n",
       "165    P413      NaN       NaN  SF11173        CHJC05   P413SF11173V1_S11   \n",
       "187    P452      NaN       NaN  SF11644        CHJC06   P452SF11644V1_S27   \n",
       "982    P532      NaN       NaN  SF13060        CHJC17  P532SF13060-V1_S29   \n",
       "\n",
       "                              file_name  processed_reads  \\\n",
       "23    p303SF10809v1_S87_R1_001.fastq.gz         26641455   \n",
       "39    P375SF11136v1_S95_R2_001.fastq.gz         30283276   \n",
       "103    p302SF10788v1_S1_R1_001.fastq.gz         28842166   \n",
       "123   p327SF10932v1_S22_R1_001.fastq.gz         30786841   \n",
       "143   p340SF11055v1_S32_R1_001.fastq.gz         34255351   \n",
       "165   p413SF11173v1_S11_R1_001.fastq.gz         45605532   \n",
       "187   p452SF11644v1_S27_R1_001.fastq.gz         57408319   \n",
       "982  P532SF13060-v1_S29_R1_001.fastq.gz         44475151   \n",
       "\n",
       "                                             full_path  \n",
       "23   /costellolab/data2/jocostello/rna_all/CH04/p30...  \n",
       "39   /costellolab/data2/jocostello/rna_all/CH04/P37...  \n",
       "103  /costellolab/data2/jocostello/rna_all/CHJC05/p...  \n",
       "123  /costellolab/data2/jocostello/rna_all/CHJC05/p...  \n",
       "143  /costellolab/data2/jocostello/rna_all/CHJC05/p...  \n",
       "165  /costellolab/data2/jocostello/rna_all/CHJC05/p...  \n",
       "187  /costellolab/data2/jocostello/rna_all/CHJC06/p...  \n",
       "982  /costellolab/data2/jocostello/rna_all/CHJC17/P...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Samples were not included in the analysis\n",
    "df_jocostello[df_jocostello['Patient'].isin(['P413','P302','P452','P375','P340','P327','P303','P532'])].drop_duplicates('Patient')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d75419cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_input_fq_files(log_file):\n",
    "    \"\"\"\n",
    "    Open a STAR Log.out file and extract the two FASTQ input filenames \n",
    "    from the line that contains '--readFilesIn'.\n",
    "    \"\"\"\n",
    "    with open(log_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if '--readFilesIn' in line:\n",
    "                # Get the part after '--readFilesIn'\n",
    "                parts = line.split('--readFilesIn', 1)\n",
    "                if len(parts) > 1:\n",
    "                    read_files_str = parts[1].strip()\n",
    "                    tokens = read_files_str.split()\n",
    "                    # Expect at least two tokens (R1 and R2)\n",
    "                    if len(tokens) >= 2:\n",
    "                        return tokens[0], tokens[1]\n",
    "    return None, None\n",
    "\n",
    "# List to accumulate rows for the new DataFrame\n",
    "rows = []\n",
    "\n",
    "for idx, row in metadata.iterrows():\n",
    "    folder = row.get('complete_path')\n",
    "    \n",
    "    # Skip rows where 'complete_path' is not a valid string\n",
    "    if not isinstance(folder, str):\n",
    "        print(f\"Skipping row {idx}: invalid complete_path {folder}\")\n",
    "        continue\n",
    "\n",
    "    # Remove the last part (the filename) to get the folder\n",
    "    folder = os.path.dirname(folder)\n",
    "    \n",
    "    # Find all Log.out files in the folder\n",
    "    log_files = glob.glob(os.path.join(folder, '*Log.out'))\n",
    "    if not log_files:\n",
    "        print(f\"No Log.out files found in folder: {folder}\")\n",
    "        continue\n",
    "\n",
    "    # Assume the first found log file is the one to use\n",
    "    r1_input, r2_input = extract_input_fq_files(log_files[0])\n",
    "    \n",
    "    # Build a new row with selected metadata and the extracted FASTQ file paths\n",
    "    new_row = {\n",
    "        'SF#': row['SF#'],\n",
    "        'arriba_filename': row['file_path_x'],\n",
    "        'patient': row['patient'],\n",
    "        'Histology': row['Histology'],\n",
    "        'Tumor': row['Tumor'],\n",
    "        'R1': r1_input,\n",
    "        'R2': r2_input\n",
    "    }\n",
    "    rows.append(new_row)\n",
    "\n",
    "# Create the new DataFrame\n",
    "new_df = pd.DataFrame(rows)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "78bf927c",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_copy = metadata.drop_duplicates('file_path_x')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "91709382",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "\n",
    "def extract_version(file_path):\n",
    "    match = re.search(r'V(\\d+)', file_path)\n",
    "    return int(match.group(1)) if match else None\n",
    "\n",
    "# Add a new column for version numbers\n",
    "metadata['version'] = metadata['file_path_x'].astype(str).apply(extract_version)\n",
    "\n",
    "# Find the maximum version per SF#\n",
    "max_versions = metadata.groupby('SF#')['version'].max().reset_index()\n",
    "max_versions['version'] = max_versions['version'].fillna(0).astype(int)\n",
    "\n",
    "# Create lookup dictionaries for file_path_x and complete_path\n",
    "lookup_dict = metadata.set_index(['SF#', 'version'])['file_path_x'].to_dict()\n",
    "complete_path_lookup = metadata.set_index(['SF#', 'version'])['complete_path'].to_dict()\n",
    "sf_lookup_dict = metadata.set_index(['SF#'])['patient'].to_dict()\n",
    "\n",
    "# Generate the complete version table per patient\n",
    "expanded_data = [\n",
    "    {\n",
    "        \"patient\": sf_lookup_dict.get(row['SF#'], None),\n",
    "        \"complete_path\": complete_path_lookup.get((row['SF#'], v), None),\n",
    "        \"SF#\": row['SF#'],\n",
    "        \"version\": int(v),\n",
    "        \"file_path_x\": lookup_dict.get((row['SF#'], v), None)\n",
    "    }\n",
    "    for _, row in max_versions.iterrows()\n",
    "    for v in range(1, int(row['version']) + 1)\n",
    "]\n",
    "\n",
    "# Create the final expanded DataFrame\n",
    "expanded_metadata = pd.DataFrame(expanded_data)\n",
    "expanded_metadata = expanded_metadata.sort_values(by=['SF#', 'version'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "50f70b31",
   "metadata": {},
   "outputs": [],
   "source": [
    "directory_path = \"/costellolab/data2/jocostello/rna_all\"\n",
    "\n",
    "# List of patients to exclude\n",
    "excluded_patients = [\n",
    "    'P300', 'P471', 'P457', 'P481', 'P458',\n",
    "    'P499', 'P500', 'P507', 'P516', 'P498',\n",
    "    'P505', 'P521', 'P524', 'P547', 'P528'\n",
    "]\n",
    "\n",
    "# Find rows where 'file_path_x' is NaN (missing)\n",
    "missing_file_paths = expanded_metadata[expanded_metadata['file_path_x'].isna()]\n",
    "\n",
    "# Remove rows for the excluded patients\n",
    "missing_file_paths = missing_file_paths[~missing_file_paths['patient'].isin(excluded_patients)]\n",
    "\n",
    "# Dictionary to store found FASTQ file paths\n",
    "found_fastq_files = {}\n",
    "\n",
    "# Iterate through each missing entry to find corresponding FASTQ files\n",
    "for index, row in missing_file_paths.iterrows():\n",
    "    patient = row['patient']\n",
    "    sf_number = row['SF#']\n",
    "    vial = row['version']\n",
    "    \n",
    "    # If complete_path is not NaN, remove the file name to get the folder path.\n",
    "    # (If complete_path is NaN, we skip this row.)\n",
    "    if pd.notna(row['complete_path']):\n",
    "        folder = os.path.dirname(row['complete_path'])\n",
    "    else:\n",
    "        continue\n",
    "\n",
    "    # Construct the search pattern for FASTQ files related to the sf_number and vial\n",
    "    search_pattern = f\"{directory_path}/*/trim*/*{sf_number}*{vial}_S*fq.gz\"\n",
    "    \n",
    "    # Use glob to find matching files\n",
    "    matching_files = glob.glob(search_pattern)\n",
    "    \n",
    "    # Create a name identifier from patient, SF#, and vial\n",
    "    name = f\"{patient}{sf_number}V{vial}\"\n",
    "    \n",
    "    # Store the results (if no files were found, None is stored)\n",
    "    found_fastq_files[name] = matching_files if matching_files else None\n",
    "\n",
    "# # Optionally, if you want to remove these rows from the original expanded_metadata DataFrame,\n",
    "# # you can do so by filtering it:\n",
    "# expanded_metadata = expanded_metadata[~expanded_metadata['patient'].isin(excluded_patients) | expanded_metadata['file_path_x'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "93d85b2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>complete_path</th>\n",
       "      <th>SF#</th>\n",
       "      <th>version</th>\n",
       "      <th>file_path_x</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P300</td>\n",
       "      <td>None</td>\n",
       "      <td>SF10711</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P300</td>\n",
       "      <td>None</td>\n",
       "      <td>SF10711</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P300</td>\n",
       "      <td>None</td>\n",
       "      <td>SF10711</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>P471</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11834</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>P458</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11913</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>P457</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11915</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>P457</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11915</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>P457</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11915</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>P457</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11915</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>P457</td>\n",
       "      <td>None</td>\n",
       "      <td>SF11915</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>P481</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12209</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>P498</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12251</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>P499</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12311</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>P499</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12311</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>P500</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12329</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>P507</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12446</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>P516</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12548</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>P516</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12548</td>\n",
       "      <td>13</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>P505</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12562</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>P520</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12625</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>P521</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12631</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>P524</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12681</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>P528</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12790</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>P528</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12790</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>P529</td>\n",
       "      <td>None</td>\n",
       "      <td>SF12794</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>P544</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13059</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>P547</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13262</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>P550</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13455</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>525</th>\n",
       "      <td>P558</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13501</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>P558</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13501</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>P558</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13501</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>P561</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13618</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>P561</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13618</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>P556</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13967</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>P557</td>\n",
       "      <td>None</td>\n",
       "      <td>SF13977</td>\n",
       "      <td>9</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    patient complete_path      SF#  version file_path_x\n",
       "1      P300          None  SF10711        2        None\n",
       "2      P300          None  SF10711        3        None\n",
       "8      P300          None  SF10711        9        None\n",
       "46     P471          None  SF11834        7        None\n",
       "72     P458          None  SF11913        3        None\n",
       "82     P457          None  SF11915        1        None\n",
       "85     P457          None  SF11915        4        None\n",
       "86     P457          None  SF11915        5        None\n",
       "88     P457          None  SF11915        7        None\n",
       "90     P457          None  SF11915        9        None\n",
       "153    P481          None  SF12209        9        None\n",
       "170    P498          None  SF12251        6        None\n",
       "188    P499          None  SF12311        4        None\n",
       "190    P499          None  SF12311        6        None\n",
       "202    P500          None  SF12329        3        None\n",
       "232    P507          None  SF12446        5        None\n",
       "276    P516          None  SF12548        9        None\n",
       "280    P516          None  SF12548       13        None\n",
       "289    P505          None  SF12562        5        None\n",
       "342    P520          None  SF12625        9        None\n",
       "349    P521          None  SF12631        6        None\n",
       "365    P524          None  SF12681        3        None\n",
       "386    P528          None  SF12790        4        None\n",
       "388    P528          None  SF12790        6        None\n",
       "399    P529          None  SF12794        7        None\n",
       "457    P544          None  SF13059        5        None\n",
       "473    P547          None  SF13262        4        None\n",
       "503    P550          None  SF13455        6        None\n",
       "525    P558          None  SF13501        1        None\n",
       "530    P558          None  SF13501        6        None\n",
       "531    P558          None  SF13501        7        None\n",
       "552    P561          None  SF13618        1        None\n",
       "556    P561          None  SF13618        5        None\n",
       "575    P556          None  SF13967        1        None\n",
       "593    P557          None  SF13977        9        None"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expanded_metadata[~(expanded_metadata['file_path_x'].notna())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e4cab20c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# P528 SF12790 v6 /costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-1_S29_R1_001.fastq.gz\n",
    "# P457 SF11915 v9 is missing /costellolab/data2/jocostello/rna_all/CHJC09/P457SF11915v9_S25_R1_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "6b0999ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "script_folder = \"/costellolab/data1/sermare/fusion_hg38/230502024/SLURM_JOBS\"\n",
    "folder = \"/costellolab/data1/sermare/fusion_hg38/230502024\"\n",
    "\n",
    "# Process each sample from the dictionary\n",
    "for sample_name, fastq_files in found_fastq_files.items():\n",
    "    print(f\"Processing: {sample_name}\")\n",
    "\n",
    "    # Get the R1 and R2 file paths\n",
    "    r1_file = fastq_files[0] if '_R1_' in fastq_files[0] else fastq_files[1]\n",
    "    r2_file = fastq_files[1] if '_R2_' in fastq_files[1] else fastq_files[0]\n",
    "\n",
    "    # Extract filenames from full paths\n",
    "    trimmed_file_1 = os.path.basename(r1_file)\n",
    "    trimmed_file_2 = os.path.basename(r2_file)\n",
    "\n",
    "    # Use the sample name directly from the dictionary key\n",
    "    sample_output = sample_name\n",
    "\n",
    "    # Create the path for the SLURM script file\n",
    "    script_file_path = os.path.join(script_folder, f\"{sample_output}.sh\")\n",
    "\n",
    "    with open(script_file_path, \"w\") as file:\n",
    "        # Write SLURM header\n",
    "        file.write(\"#!/bin/bash\\n\")\n",
    "        file.write(f\"#SBATCH --job-name={sample_output}\\n\")\n",
    "        file.write(\"#SBATCH --nodes=1\\n\")\n",
    "        file.write(\"#SBATCH --partition=costellolab\\n\")\n",
    "        file.write(\"#SBATCH --ntasks=1\\n\")\n",
    "        file.write(\"#SBATCH --mem=50G\\n\")\n",
    "        file.write(\"#SBATCH --time=99:00:00\\n\")\n",
    "        file.write(f\"#SBATCH --output={script_folder}/ERROR/%x-%j.out\\n\")\n",
    "        file.write(f\"#SBATCH --error={script_folder}/error_%j.err\\n\")\n",
    "        file.write(\"#SBATCH --mail-type=END,FAIL\\n\")\n",
    "        file.write(\"#SBATCH --mail-user=sergiomar10@berkeley.edu\\n\\n\")\n",
    "\n",
    "        # Change to the fusions directory and load required modules\n",
    "        file.write(f\"cd {folder}\\n\")\n",
    "        file.write(\"module load CBI\\n\")\n",
    "        file.write(\"module load WitteLab\\n\")\n",
    "        file.write(\"module load python3\\n\\n\")\n",
    "        file.write(\"source /costellolab/data3/sermare/miniconda3/etc/profile.d/conda.sh\\n\")\n",
    "        file.write(\"conda activate Arriba\\n\\n\")\n",
    "\n",
    "        # Create and move into the sample-specific output directory\n",
    "        file.write(f\"mkdir -p {folder}/{sample_output}\\n\")\n",
    "        file.write(f\"cd {folder}/{sample_output}\\n\\n\")\n",
    "\n",
    "        # Build the STAR and Arriba command pipeline\n",
    "        star_command = (\n",
    "            \"STAR \"\n",
    "            \"--runThreadN 8 \"\n",
    "            \"--genomeDir /costellolab/data3/sermare/arriba_hg38/STAR_index_hg38_GENCODE38 --genomeLoad NoSharedMemory \"\n",
    "            f\"--readFilesIn {r1_file} {r2_file} --readFilesCommand zcat \"\n",
    "            \"--outStd BAM_Unsorted --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 \"\n",
    "            \"--outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 \"\n",
    "            \"--alignSJstitchMismatchNmax 5 -1 5 5 \"\n",
    "            \"--chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 \"\n",
    "            \"--chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50 | \"\n",
    "            \"arriba \"\n",
    "            \"-x /dev/stdin \"\n",
    "            f\"-o {sample_output}_fusions.tsv -O {sample_output}_fusions.discarded.tsv \"\n",
    "            \"-a /costellolab/data3/sermare/arriba_hg38/hg38.fa -g /costellolab/data3/sermare/arriba_hg38/GENCODE38.gtf \"\n",
    "            \"-b /costellolab/data3/sermare/arriba_v2.3.0/database/blacklist_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-k /costellolab/data3/sermare/arriba_v2.3.0/database/known_fusions_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-t /costellolab/data3/sermare/arriba_v2.3.0/database/known_fusions_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-p /costellolab/data3/sermare/arriba_v2.3.0/database/protein_domains_hg38_GRCh38_v2.3.0.gff3\"\n",
    "        )\n",
    "        file.write(star_command + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a8b5afd8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# P528 SF12790 v6 /costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-1_S29_R1_001.fastq.gz\n",
    "# P457 SF11915 v9 is missing /costellolab/data2/jocostello/rna_all/CHJC09/P457SF11915v9_S25_R1_001.fastq.gz\n",
    "found_fastq_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2f00f4ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "found_fastq_files_other = {'P528SF12790-v6-1': ['/costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-1_S29_R1_001.fastq.gz',\n",
    " '/costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-1_S29_R2_001.fastq.gz'], \n",
    "                    'P528SF12790-v6-2' : ['/costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-2_S30_R1_001.fastq.gz',\n",
    " '/costellolab/data2/jocostello/rna_all/CHJC26/P528SF12790-v6-2_S30_R2_001.fastq.gz'], \n",
    "                    'P457SF11915v9':['/costellolab/data2/jocostello/rna_all/CHJC09/P457SF11915v9_S25_R1_001.fastq.gz',\n",
    " '/costellolab/data2/jocostello/rna_all/CHJC09/P457SF11915v9_S25_R2_001.fastq.gz']}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d817270a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: P528SF12790-v6-1\n",
      "Processing: P528SF12790-v6-2\n",
      "Processing: P457SF11915v9\n"
     ]
    }
   ],
   "source": [
    "script_folder = \"/costellolab/data1/sermare/fusion_hg38/230502024/SLURM_JOBS\"\n",
    "folder = \"/costellolab/data1/sermare/fusion_hg38/230502024\"\n",
    "\n",
    "# Process each sample from the dictionary\n",
    "for sample_name, fastq_files in found_fastq_files_other.items():\n",
    "    print(f\"Processing: {sample_name}\")\n",
    "\n",
    "    # Get the R1 and R2 file paths\n",
    "    r1_file = fastq_files[0] if '_R1_' in fastq_files[0] else fastq_files[1]\n",
    "    r2_file = fastq_files[1] if '_R2_' in fastq_files[1] else fastq_files[0]\n",
    "\n",
    "    # Extract filenames from full paths\n",
    "    trimmed_file_1 = os.path.basename(r1_file)\n",
    "    trimmed_file_2 = os.path.basename(r2_file)\n",
    "\n",
    "    # Use the sample name directly from the dictionary key\n",
    "    sample_output = sample_name\n",
    "\n",
    "    # Create the path for the SLURM script file\n",
    "    script_file_path = os.path.join(script_folder, f\"{sample_output}.sh\")\n",
    "\n",
    "    with open(script_file_path, \"w\") as file:\n",
    "        # Write SLURM header\n",
    "        file.write(\"#!/bin/bash\\n\")\n",
    "        file.write(f\"#SBATCH --job-name={sample_output}\\n\")\n",
    "        file.write(\"#SBATCH --nodes=1\\n\")\n",
    "        file.write(\"#SBATCH --partition=costellolab\\n\")\n",
    "        file.write(\"#SBATCH --ntasks=1\\n\")\n",
    "        file.write(\"#SBATCH --mem=50G\\n\")\n",
    "        file.write(\"#SBATCH --time=99:00:00\\n\")\n",
    "        file.write(f\"#SBATCH --output={script_folder}/ERROR/%x-%j.out\\n\")\n",
    "        file.write(f\"#SBATCH --error={script_folder}/error_%j.err\\n\")\n",
    "        file.write(\"#SBATCH --mail-type=END,FAIL\\n\")\n",
    "        file.write(\"#SBATCH --mail-user=sergiomar10@berkeley.edu\\n\\n\")\n",
    "\n",
    "        # Change to the fusions directory and load required modules\n",
    "        file.write(f\"cd {folder}\\n\")\n",
    "        file.write(\"module load CBI\\n\")\n",
    "        file.write(\"module load WitteLab\\n\")\n",
    "        file.write(\"module load python3\\n\\n\")\n",
    "        file.write(\"source /costellolab/data3/sermare/miniconda3/etc/profile.d/conda.sh\\n\")\n",
    "        file.write(\"conda activate Arriba\\n\\n\")\n",
    "\n",
    "        # Create and move into the sample-specific output directory\n",
    "        file.write(f\"mkdir -p {folder}/{sample_output}\\n\")\n",
    "        file.write(f\"cd {folder}/{sample_output}\\n\\n\")\n",
    "\n",
    "        # Build the STAR and Arriba command pipeline\n",
    "        star_command = (\n",
    "            \"STAR \"\n",
    "            \"--runThreadN 8 \"\n",
    "            \"--genomeDir /costellolab/data3/sermare/arriba_hg38/STAR_index_hg38_GENCODE38 --genomeLoad NoSharedMemory \"\n",
    "            f\"--readFilesIn {r1_file} {r2_file} --readFilesCommand zcat \"\n",
    "            \"--outStd BAM_Unsorted --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 \"\n",
    "            \"--outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 \"\n",
    "            \"--alignSJstitchMismatchNmax 5 -1 5 5 \"\n",
    "            \"--chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 \"\n",
    "            \"--chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50 | \"\n",
    "            \"arriba \"\n",
    "            \"-x /dev/stdin \"\n",
    "            f\"-o {sample_output}_fusions.tsv -O {sample_output}_fusions.discarded.tsv \"\n",
    "            \"-a /costellolab/data3/sermare/arriba_hg38/hg38.fa -g /costellolab/data3/sermare/arriba_hg38/GENCODE38.gtf \"\n",
    "            \"-b /costellolab/data3/sermare/arriba_v2.3.0/database/blacklist_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-k /costellolab/data3/sermare/arriba_v2.3.0/database/known_fusions_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-t /costellolab/data3/sermare/arriba_v2.3.0/database/known_fusions_hg38_GRCh38_v2.3.0.tsv.gz \"\n",
    "            \"-p /costellolab/data3/sermare/arriba_v2.3.0/database/protein_domains_hg38_GRCh38_v2.3.0.gff3\"\n",
    "        )\n",
    "        file.write(star_command + \"\\n\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biopython",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
